ospf6d/ospf6_top.c

```diff
@@ -652,21 +681,31 @@ DEFUN(no_ospf6_router_id,
 {
 	VTY_DECLVAR_CONTEXT(ospf6, o);
 	struct ospf6_area *oa;
-	struct listnode *node;
+	struct listnode *node, *nnode;
 
 	o->router_id_static = 0;
 
+	if (o->router_id == 0)
+		return CMD_SUCCESS;
+
 	for (ALL_LIST_ELEMENTS_RO(o->area_list, node, oa)) {
 		if (oa->full_nbrs) {
 			vty_out(vty,
 				"For this router-id change to take effect, save config and restart ospf6d\n");
 			return CMD_SUCCESS;
 		}
 	}
+
+	for (ALL_LIST_ELEMENTS(o->area_list, node, nnode, oa))
+		ospf6_area_disable(oa);
```

Comment:
Only area disable won't be enough. We need to reset even process level and interface level dynamic datas as well.

Reply:
The fixes have been tested and it works as expected.

> ospf6_db_clear(ospf6);
This function does not exist (at least on the latest master)

> ospf6_flush_self_originated_lsas_now(ospf6);
> ospf6_db_clear(ospf6);
> ospf6_asbr_redistribute_reset(ospf6);
> ospf6_interface_clear(vty, ifp);

It seems that `ospf6_area_disable` already flushes the lsdb and disables the interfaces.

I would only add ospf6_asbr_redistribute_reset(ospf6);

```
void ospf6_area_disable(struct ospf6_area *oa)
{
	struct listnode *node, *nnode;
	struct ospf6_interface *oi;

	UNSET_FLAG(oa->flag, OSPF6_AREA_ENABLE);

	for (ALL_LIST_ELEMENTS(oa->if_list, node, nnode, oi))
		ospf6_interface_disable(oi);

	ospf6_abr_disable_area(oa);
	ospf6_lsdb_remove_all(oa->lsdb);
	ospf6_lsdb_remove_all(oa->lsdb_self);

	ospf6_spf_table_finish(oa->spf_table);
	ospf6_route_remove_all(oa->route_table);

	THREAD_OFF(oa->thread_router_lsa);
	THREAD_OFF(oa->thread_intra_prefix_lsa);
}

```

Reply:
```
	ospf6_lsdb_remove_all(oa->lsdb);
	ospf6_lsdb_remove_all(oa->lsdb_self);
```
If lsdb is flushed, what is the need for ospf6_asbr_redistribute_reset ?

Reply:
This will be required for origination of external LSAs

Reply:
Zebra will send the routes again to ospf6d when you call this api.

Reply:
> This will be required for origination of external LSAs

All LSAs are supposed to be flushed

> Zebra will send the routes again to ospf6d when you call this api.

OK I will also test redistribution from route to ospfv3

Reply:
> ospf6_flush_self_originated_lsas_now(ospf6);
> ospf6_interface_clear(vty, ifp);

Do you still need that ?